node-manager/src/node.rs

```diff
@@ -365,6 +367,114 @@ impl Node {
         Ok(invoice)
     }
 
+    pub fn get_invoice(&self, invoice_str: String) -> Result<MutinyInvoice, MutinyError> {
+        let invoice = Invoice::from_str(&invoice_str).map_err(Into::<MutinyError>::into)?;
+        let payment_hash = invoice.payment_hash();
+        let (payment_info, inbound) = self.get_payment_info_from_persisters(payment_hash)?;
+        let mut mutiny_invoice: MutinyInvoice = invoice.into();
+        mutiny_invoice.is_send = !inbound;
+        mutiny_invoice.paid = matches!(payment_info.status, HTLCStatus::Succeeded);
+        Ok(mutiny_invoice)
+    }
+
+    pub fn list_invoices(&self) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let mut inbound_invoices = self.list_payment_info_from_persisters(true);
+        let mut outbound_invoices = self.list_payment_info_from_persisters(false);
+        inbound_invoices.append(&mut outbound_invoices);
+        Ok(inbound_invoices)
+    }
+
+    fn list_payment_info_from_persisters(&self, inbound: bool) -> Vec<MutinyInvoice> {
+        self.persister
+            .list_payment_info(inbound)
+            .into_iter()
+            .filter_map(|(h, i)| match i.bolt11 {
+                Some(bolt11) => {
+                    // Construct an invoice from a bolt11, easy
+                    let invoice_res = Invoice::from_str(&bolt11).map_err(Into::<MutinyError>::into);
+                    match invoice_res {
+                        Ok(invoice) => {
+                            let mut mutiny_invoice: MutinyInvoice = invoice.into();
+                            mutiny_invoice.is_send = !inbound;
+                            mutiny_invoice.paid = matches!(i.status, HTLCStatus::Succeeded);
+                            Some(mutiny_invoice)
+                        }
+                        Err(_) => None,
+                    }
+                }
+                None => {
+                    // Constructing MutinyInvoice from no invoice, harder
+                    let paid = matches!(i.status, HTLCStatus::Succeeded);
+                    // TODO finish this properly
+                    let mutiny_invoice =
+                        MutinyInvoice::new(None, None, h, None, None, 0, paid, None, !inbound);
```

Comment:
we should have `fees_paid` and `amount_sats` from payment info

Reply:
parsing them now, along with the rest of what i can parse